AstSemantics.md

```diff
@@ -256,14 +252,25 @@ one can arrange `block`s to put labels wherever one wants to jump to, except
 for one restriction: one can't jump into the middle of a loop from outside
 it. This restriction ensures the well-structured property discussed below.
 
-A `tableswitch` has a zero-based array of targets, a "default" target, an index
-operand, and a list of `case` nodes. It jumps to the target indexed in the
-array, or the default if the index is out of bounds. `tableswitch` targets may
-be either labels or `case` nodes.
+### Yielding values from control constructs
+
+The `if`, `br`, `br_if`, `case`, and `return` constructs do not yield values.
```

Comment:
IIUC, `br` and `return` can be thought of as yielding *any* value type.  That's rather different than `if` and `br_if` which have a void return type (or, using the above definition, "empty sequence of return value types") since you *can* write `(set_local $i (br $foo))` but not `(set_local $i (if ...))`.

Also, `case` is a rather weird case (pun!): I don't really consider it a first-class AST node, but rather an extension of the `tableswitch` (even if `case` does end up getting an opcode in the binary format, you could think of it as an in-order encoding of the parent `tableswitch`).  Perhaps we could fold `case` into `tableswitch` in the list above and remove it from this list here?